app/components/smart-overflow.js

```diff
@@ -0,0 +1,27 @@
+import Ember from 'ember';
+
+const { Component } = Ember;
+
+export default Component.extend({
+  classNames: ['smart-overflow'],
+  didInsertElement() {
+    this._super(...arguments);
+    var $headerSpan = this.$('div > span');
+    var $header = this.$('div');
```

Comment:
I have made changes in the code to accommodate such that it will work fine for all without removing div.